lib/discourse_activity_pub/actor_handler.rb

```diff
@@ -249,7 +243,12 @@ def valid_actor_opts?
     end
 
     def invalid_opt(key)
-      add_error(I18n.t("discourse_activity_pub.actor.warning.#{key}"))
+      opts = {}
+      if key == "invalid_username"
+        opts[:min_length] = SiteSetting.min_username_length
+        opts[:max_length] = SiteSetting.max_username_length
+      end
+      add_error(I18n.t("discourse_activity_pub.actor.warning.#{key}", opts))
```

Comment:
@angusmcleod should the plugin also account for `username change period` in core? Not sure if that raises an error, and note that the setting also blocks username changes entirely if set to 0.

Reply:
The `username` changes handled by the ActorHandler will be unaffected by `username_change_period`, as changes are not filtered through the UserGaurdian that applies it. Even though we leverage some username logic for the ActivityPub username, e.g. UsernameValidator, I don't think it'd make sense to apply the UserGuardian, or restrictions like `username_change_period`  to actor changes, as different considerations are in play. We may restrict Actor username changes, but it will be for different reasons, e.g. that a plurality of other ActivityPub services do not support changes beyond a certain period. To put it another way, the considerations pertinent to Actors are sometimes overlapping, but mostly different from the considerations pertinent to Discourse users, so I think it's best to keep them separate (which is also why we have a separate model for them).